tools/pubsys/src/repo/fetch_variant/mod.rs

```diff
@@ -0,0 +1,240 @@
+    #[arg(long)]
+    /// Path to root.json for this repo
+    root_role_path: PathBuf,
+
+    #[arg(long)]
+    /// Where to store the downloaded img files
+    outdir: PathBuf,
+
+    #[arg(long)]
+    /// The variant name without extension
+    filename_prefix: String,
+
+    #[arg(long)]
+    /// The root directory of the buildsys
+    variant_manifest: PathBuf,
+}
+
+/// Download targets
+async fn handle_download(
+    repository: &Repository,
+    outdir: &Path,
+    raw_names: &[String],
+) -> Result<(), Error> {
+    let target_names: Result<Vec<TargetName>, Error> = raw_names
+        .iter()
+        .map(|s| TargetName::new(s).context(error::InvalidTargetNameSnafu))
+        .collect();
+    let target_names = target_names?;
+    let download_target = |name: TargetName| async move {
+        println!("\t-> {}", name.raw());
+        repository
+            .save_target(&name, outdir, Prefix::None)
+            .await
+            .context(error::MetadataSnafu)
+    };
+
+    // copy requested targets, or all available targets if not specified
+    let targets: Vec<TargetName> = if target_names.is_empty() {
+        repository
+            .targets()
+            .signed
+            .targets
+            .keys()
+            .cloned()
+            .collect()
+    } else {
+        target_names
+    };
+
+    println!("Downloading targets to {outdir:?}");
+    tokio::fs::create_dir_all(outdir)
+        .await
+        .context(error::DirCreateSnafu { path: outdir })?;
+    for target in targets {
+        download_target(target).await?;
+    }
+    Ok(())
+}
+
+/// Fetch the variant and download its image targets
+async fn fetch_variant(
+    root_role_path: &PathBuf,
+    metadata_url: Url,
+    targets_url: &Url,
+    outdir: &PathBuf,
+    filename_prefix: &str,
+    variant_manifest: &PathBuf,
+    variant: &str,
+) -> Result<(), Error> {
+    // Load the repository
+    let repo = RepositoryLoader::new(
+        &repo::root_bytes(root_role_path).await?,
+        metadata_url.clone(),
+        targets_url.clone(),
+    )
+    .load()
+    .await
+    .context(repo_error::RepoLoadSnafu {
+        metadata_base_url: metadata_url.clone(),
+    })?;
+
+    // To help ensure that downloads are safe, we require that the outdir does not exist.
+    ensure!(
+        !outdir.exists(),
+        error::DownloadOutdirExistsSnafu { path: outdir }
+    );
+
+    let manifest_info = ManifestInfo::new(variant_manifest).context(error::ManifestParseSnafu)?;
+
+    let image_layout = manifest_info
+        .image_layout()
+        .context(error::MissingImageLayoutSnafu { variant })?;
+    let image_format = manifest_info.image_format();
+    let image_ext = match image_format {
+        Some(ImageFormat::Raw) | None => "img.lz4",
+        Some(ImageFormat::Qcow2) => "qcow2",
+        Some(ImageFormat::Vmdk) => "ova",
+    };
+
+    // TODO - remove this after testing
+    let clean_filename_prefix = filename_prefix
+        .strip_suffix("-dirty")
```

Comment:
If we publish images from a dirty checkout into a repo, do they not retain the suffix? It doesn't seem like we should strip it.